devito/dse/transformer.py

```diff
@@ -49,18 +49,18 @@ def rewrite(clusters, mode='advanced'):
 
     if mode is None or mode == 'noop':
         return clusters
-    elif mode not in dse_registry:
-        warning("Unknown rewrite mode(s) %s" % mode)
-        return clusters
+
+    try:
```

Comment:
maybe it's better if this whole try-except goes right below the big comment below

Reply:
Agree, But looking at the big-comment, is there any need for rewriting it... Is it easy for someone to understand when we deal with a sparse or dense? How/When/Why... It is probably just me that did not yet observe any info about sparse/dense rewriters...

Reply:
I mean to move it below? not sure I follow.

The comment below says something, that something is partly about what happens in this try-except